### DISCUSSION TITLE
how do we use"implies" for binary set judgement？

### DISCUSSION THREAD

**Discussion by lemonAriel** [yalmip/YALMIP#1110] [Problems using YALMIP]
Dear professor, I have recently been exposed to Yalmip while met some problems with the application of the ''implies function'' when using it. If I want to traverse the set of binary variables here, e.g, if s(v,j) and s(v, I) are equal to 1, then uu=1, and the following inequation is true. How should such a relationship be achieved?

```
s = binvar(v,trip,'full');
uu=binvar(1);
for v1=1:v
    for i=1:trip-1
        for j=2:trip
         C=[C,uu>=s(v1,j)+s(v1,i)-1,uu<=s(v1,j),uu<=s(v1,i)];
         C=[C,implies(uu,s(v1,j)*st(j)-s(v1,i)*(st(i)+RT(i))>=0)]; 
        end
    end   
end
```

**Comment by johanlofberg** [comment]
I don't understand the question,   **if s(v,j) and s(v, I) are equal to 1, then uu=1** is simply

`C=[C,uu>=s(v1,j)+s(v1,i)-1];`

(you have added constraints to encode if-and-only-if)

Then you have some stuff where I don't know what you are are doing. However, if your encoding of if-and-only-if was intentional, you know that **s** is 1 in that constraint when it is activated, so it is just a cumbersome (possibly nonlinear if **st** or **RT** are decision variables) way of writing 

`C=[C, implies(uu, st(j)-(st(i)+RT(i))>=0)]; `

(assuming you have kept the if-and-only-if constraints)

**Reply by lemonAriel** [reply]
Thank you, professor! i will try it!
